src/lib.rs

```diff
@@ -851,103 +854,272 @@ pub struct RegisteredClaims {
+    ///Validates that the token contains the claims defined as required
+    pub fn validate_claim_presence(&self, options: ClaimPresenceOptions) -> Result<(), ValidationError> {
+        use Presence::Required;
 
-        if options.issued_at_required && self.issued_at.is_none() {
-            Err(ValidationError::MissingRequired("iat".to_string()))?;
+        let mut missing_claims: Vec<&str> = vec![];
+
+        if options.expiry == Required && self.expiry.is_none() {
+            missing_claims.push("exp");
         }
 
-        if options.expiry_required && self.expiry.is_none() {
-            Err(ValidationError::MissingRequired("exp".to_string()))?;
+        if options.not_before == Required && self.not_before.is_none() {
+            missing_claims.push("nbf");
         }
 
-        if options.not_before_required && self.not_before.is_none() {
-            Err(ValidationError::MissingRequired("nbf".to_string()))?;
+        if options.issued_at == Required && self.issued_at.is_none() {
+            missing_claims.push("iat");
         }
 
-        let now = match options.now {
-            None => Utc::now(),
-            Some(now) => now,
-        };
+        if options.audience == Required && self.audience.is_none() {
+            missing_claims.push("aud");
+        }
 
-        let e = match options.epsilon {
-            None => std::time::Duration::from_secs(0),
-            Some(e) => e,
-        };
+        if options.issuer == Required && self.issuer.is_none() {
+            missing_claims.push("iss");
+        }
 
-        if self.expiry.is_some() && !Self::is_after(*self.expiry.unwrap(), now, e)? {
-            Err(ValidationError::TemporalError("Token expired".to_string()))?;
+        if options.subject == Required && self.subject.is_none() {
+            missing_claims.push("sub");
         }
 
-        if self.issued_at.is_some() && !Self::is_before(*self.issued_at.unwrap(), now, e)? {
-            Err(ValidationError::TemporalError("Token issued in the future".to_string()))?;
+        if options.id == Required && self.id.is_none() {
+            missing_claims.push("jti");
         }
 
-        if self.not_before.is_some() && !Self::is_before(*self.not_before.unwrap(), now, e)? {
-            Err(ValidationError::TemporalError("Token not valid yet".to_string()))?;
+        if missing_claims.len() == 0 {
+            Ok(())
+        } else {
+            Err(ValidationError::MissingRequiredClaims(
+                missing_claims.into_iter().map(|v| v.into()).collect(),
+            ))
         }
+    }
 
-        Ok(())
+    ///Validates that if the token has an `exp` claim, it has not passed.
+    pub fn validate_exp(&self, validation: Validation<TemporalOptions>) -> Result<(), ValidationError> {
+        match validation {
+            Validation::Ignored => Ok(()),
+            Validation::Validate(temporal_options) => {
+                let now = temporal_options.now.unwrap_or(Utc::now());
```

Comment:
Use [`unwrap_or_else`](https://doc.rust-lang.org/std/option/enum.Option.html#method.unwrap_or_else) for lazy evaluation.

Reply:
Is this such an expensive operation that it's worth it?